src/Data/Sum/Relation/Unary/All.agda

```diff
@@ -0,0 +1,32 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Heterogeneous All-predicate for disjoint sums
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Sum.Relation.Unary.All where
+
+open import Data.Sum       using (_⊎_); open _⊎_
+open import Level          using (Level; _⊔_)
+open import Relation.Unary using (Pred)
+
+private
+  variable
+    a b c p q : Level
+    A B : Set _
+
+data All₂ {A : Set a} {B : Set b} (P : Pred A p) (Q : Pred B q)
+       : Pred (A ⊎ B) (a ⊔ b ⊔ p ⊔ q) where
+  inj₁ : ∀ {a} → P a → All₂ P Q (inj₁ a)
+  inj₂ : ∀ {b} → Q b → All₂ P Q (inj₂ b)
+
+-- Elimination
+
+[_,_] : ∀ {P : Pred A p} {Q : Pred B q} {C : (x : A ⊎ B) → All₂ P Q x → Set c} →
```

Comment:
Minor thing, but FYI our new variable policy, outlined in the style guide, allows one to use variables over `P` and `Q` in this file as well.